xmtp_mls/src/groups/commit_log.rs

```diff
@@ -344,46 +316,248 @@ where
+            let commit_log_entry: &CommitLogEntry = entry;
+            let log_entry = PlaintextCommitLogEntry::decode(
+                commit_log_entry.serialized_commit_log_entry.as_slice(),
+            )?;
+
+            // From the stored remote commit log, fetch the following info:
+            // 1. The latest applied epoch authenticator
+            // 2. The latest applied epoch number
+            // 3. The latest stored sequence id
+            let validation_info = conn.get_remote_log_validation_info(&group_id)?;
+            if self.should_skip_remote_commit_log_entry(&validation_info, &log_entry) {
+                continue;
+            }
+
+            num_entries_saved += 1;
+            NewRemoteCommitLog {
+                log_sequence_id: commit_log_entry.sequence_id as i64,
                 group_id: log_entry.group_id,
                 commit_sequence_id: log_entry.commit_sequence_id as i64,
                 commit_result: CommitResult::from(
                     ProtoCommitResult::try_from(log_entry.commit_result)
                         .unwrap_or(ProtoCommitResult::Unspecified),
                 ),
-                applied_epoch_number: Some(log_entry.applied_epoch_number as i64),
-                applied_epoch_authenticator: Some(log_entry.applied_epoch_authenticator),
+                applied_epoch_number: log_entry.applied_epoch_number as i64,
+                applied_epoch_authenticator: log_entry.applied_epoch_authenticator,
             }
             .store(conn)?;
-            if entry.sequence_id > latest_download_cursor {
-                latest_download_cursor = entry.sequence_id;
-            }
-            if log_entry.commit_sequence_id > latest_sequence_id {
-                latest_sequence_id = log_entry.commit_sequence_id;
+            if log_entry.commit_sequence_id as i64 > latest_commit_sequence_id {
+                latest_commit_sequence_id = log_entry.commit_sequence_id as i64;
             }
         }
-        conn.update_cursor(
-            &group_id,
-            xmtp_db::refresh_state::EntityKind::CommitLogDownload,
-            latest_download_cursor as i64,
-        )?;
+        if let Some(last_entry) = commit_log_response.commit_log_entries.last() {
+            latest_download_cursor = last_entry.sequence_id as i64;
+            conn.update_cursor(
+                &group_id,
+                xmtp_db::refresh_state::EntityKind::CommitLogDownload,
+                latest_download_cursor,
+            )?;
+        }
+
         Ok(UpdateCursorsResult {
             conversation_id: group_id,
-            last_entry_saved_commit_sequence_id: latest_sequence_id as i64,
-            last_entry_saved_remote_log_sequence_id: latest_download_cursor as i64,
+            num_entries_saved,
+            last_entry_saved_commit_sequence_id: latest_commit_sequence_id,
+            last_entry_saved_remote_log_sequence_id: latest_download_cursor,
```

Comment:
Are these two fields used anywhere? I'm wondering if we can get away with only returning `num_entries_saved` and nothing else, but I definitely could have misunderstood

Reply:
removed 07a28084188ecc7ac7c0f5f5614227211476b4e4